lib/BannerPlugin.js

```diff
@@ -78,13 +90,15 @@ class BannerPlugin {
 							basename = filename.substr(lastSlashIndex + 1);
 						}
 
-						const comment = compilation.getPath(banner, {
+						const data = {
```

Comment:
I'm seeing an error when trying out the feature, see #7269.

Also, would it make sense to expose the same variable names as the string placeholders?

From the docs:

## Placeholders

Since webpack 2.5.0, placeholders are evaluated in the `banner` string:

```javascript
new webpack.BannerPlugin({
  banner: "hash:[hash], chunkhash:[chunkhash], name:[name], filebase:[filebase], query:[query], file:[file]"
})
```

I can see how it'd be useful to have access to everything in `chunk` for example, but just to keep the future docs more straightforward, maybe the exposed variables could be consistent.